docs/LoongArch-ELF-ABI-EN.adoc

```diff
@@ -645,6 +645,205 @@ with check 32-bit unsigned overflow
 |R_LARCH_GNU_VTENTRY
 |GNU C++ vtable member usage
 |
+
+4+|... Reserved
+
+|64
+|R_LARCH_B16
+|18-bit PC-relative jump
+|`+(*(uint32_t *) PC) [25 ... 10] = (S+A-PC) [17 ... 2]+`
+
+with check 18-bit signed overflow and 4-bit aligned
+
+|65
+|R_LARCH_B21
+|23-bit PC-relative jump
+|`+(*(uint32_t *) PC) [4 ... 0] = (S+A-PC) [22 ... 18],+`
+
+`+(*(uint32_t *) PC) [25 ... 10] = (S+A-PC) [17 ... 2]+`
+
+with check 23-bit signed overflow and 4-bit aligned
+
+|66
+|R_LARCH_B26
+|28-bit PC-relative jump
+|`+(*(uint32_t *) PC) [9 ... 0] = (S+A-PC) [27 ... 18],+`
+
+`+(*(uint32_t *) PC) [25 ... 10] = (S+A-PC) [17 ... 2]+`
+
+with check 28-bit signed overflow and 4-bit aligned
+
+|67
+|R_LARCH_ABS_HI20
+| [31 ... 12] bits of 32/64-bit absolute address
+|`+(*(uint32_t *) PC) [24 ... 5] = (S+A) [31 ... 12]+`
+
+|68
+|R_LARCH_ABS_LO12
+|[11 ... 0] bits of 32/64-bit absolute address
+|`+(*(uint32_t *) PC) [21 ... 10] = (S+A) [11 ... 0]+`
+
+|69
+|R_LARCH_ABS64_LO20
+|[51 ... 32] bits of 64-bit absolute address
+|`+(*(uint32_t *) PC) [24 ... 5] = (S+A) [51 ... 32]+`
+
+|70
+|R_LARCH_ABS64_HI12
+|[63 ... 52] bits of 64-bit absolute address
+|`+(*(uint32_t *) PC) [21 ... 10] = (S+A) [63 ... 52]+`
+
+|71
+|R_LARCH_PCALA_HI20
+|[31 ... 12] bits of 32/64-bit PC-relative offset
+|`+(*(uint32_t *) PC) [24 ... 5] = (((S+A) & ~0xfff) - (PC & ~0xfff)) [31 ... 12]+`
```

Comment:
```suggestion
|[31 ... 12] bits of 32/64-bit PC-relative offset to the nearest 4KB bounary for the target
|`+(*(uint32_t *) PC) [24 ... 5] = (((S+A+0x800) & ~0xfff) - (PC & ~0xfff)) [31 ... 12]+`
```

For example, S+A = 0x12001800 and PC = 0x12000010.  We want:

```
pcalau12i $t0, 0x2   # 0x2000 >> 12
addi.d $t0, $t0, 0x800 # -2048
```

to get S+A.  Without `+0x800` we get `pcalau12i $t0, $0x1` then we can't use `addi.d` (`ori` is needed for a positive offset >= 2048).  But the binutils patch uses `addi.d`, not `ori`.

Reply:
pcalau12i $t0, 0x2, (S+A+0x800)&(~0xfff) - (PC & (~0xfff))